shell/mixins/resource-fetch-api-pagination.js

```diff
@@ -0,0 +1,237 @@
+import {
```

Comment:
I don't know if we want to continue adding new mixins

![image](https://github.com/rancher/dashboard/assets/55104481/6fa69e33-db87-4a68-aa32-59edaa753f0a)

Reply:
I was 50:50 on this. Went with a mixin as there's an existing set and pattern for ResourceList. These are all tightly coupled and depend on each other.
- `resource-fetch`
  - contains the route $fetchType which makes api requests used in ResourceList and custom list components. This references the other mixins and is used by them to make that request.
- `resource-fetch-namespaced`
  - supplies functionality for `Require Namespace / Project Filtering` to sibling mixins and parent components
- `resource-fetch-api-pagination` (new one)
  - supplies functionality for pagination to sibling mixins and parent components
  
When we get to the point no incremental loading, manual refresh or required ns / filtering is needed we can get rid of the existing mixins and port only the pagination feature over